pkg/traceql/lexer.go

```diff
@@ -95,30 +99,48 @@ func (l *lexer) Lex(lval *yySymType) int {
 		return END_ATTRIBUTE
 	}
 
-	r := l.Scan()
-
 	// if we are currently parsing an attribute then just grab everything until we find a character that ends the attribute.
 	// we will handle parsing this out in ast.go
 	if l.parsingAttribute {
-		str := l.TokenText()
-		// parse out any scopes here
-		tok := tokens[str+string(l.Peek())]
+		// parse out any scopes here, but copy it so we dont move main scanner if it's not a scope
+		s := l.Scanner
+		s.Scan()
+		str := s.TokenText()
+		tok := tokens[str+string(s.Peek())]
 		if tok == RESOURCE_DOT || tok == SPAN_DOT {
+			for i := 0; i < len(str); i++ {
```

Comment:
what are these lines for? does a single Next() not consume the token anymore?

Reply:
copying  scanner here so it does not move the main scanner unless it finds `RESOURCE_DOT / SPAN_DOT`.

This is to avoid scenarios like `{ ."foo".bar = "val"}` calling Scan() on main scanner after reading the first dot would move it to second dot.